hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1573,7 +1560,7 @@ protected S3AReadOpContext createReadContext(
         statistics,
         statisticsContext,
         fileStatus,
-        futurePool,
+        unboundedThreadPool,
```

Comment:
Why not the `boundedThreadPool` originally wrapped by `futurePool`?

Reply:
that's my bad, it should be `boundedThreadPool`. Have updated